grammar.js

```diff
@@ -1,157 +1,165 @@
-const PREC = {
-  PATH: 10, // foo/bar
-  BIND: 5, // foo:bar
-};
-
-const SYMBOL_HEAD =
-  /[^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9]/;
-
-const SYMBOL_TAIL =
-  choice(SYMBOL_HEAD, /[.0-9]/);
-
-const SYMBOL =
-  token(seq(SYMBOL_HEAD, repeat(SYMBOL_TAIL)));
-
-const SUBPATH =
-  repeat1(seq("/", optional(SYMBOL)));
+/**
+ * @file Bass grammar for tree-sitter
+ * @author Alex Suraci <suraci.alex@gmail.com>
+ * @author Amaan Qureshi <amaanq12@gmail.com>
+ * @license MIT
+ * @see {@link https://bass-lang.org/|official website}
+ */
+
+/* eslint-disable arrow-parens */
+/* eslint-disable camelcase */
+/* eslint-disable-next-line spaced-comment */
+/// <reference types="tree-sitter-cli/dsl" />
+// @ts-check
+
+const SYMBOL = token(
+  /[^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9][^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9.]*/,
```

Comment:
```suggestion
  /[^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:.0-9][^\t\n\v\f\r \u0085\u00A0,"()\[\]{};^/:]*/,
```

`.` and digits are valid in a symbol/keyword, just not at the start:

```clojure
❯ bass
=> :foo123.bar
foo123.bar
```

Apparently they're allowed at the start when using a keyword literal to construct a symbol though:

```clojure
❯ bass
=> :.foo
.foo
=> :0foo
0foo
```

I haven't decided whether that's intentional so feel free to skip that :laughing:

Reply:
Uhh, are periods not a sort of delimiter for a sub symbol/type? I really dk bass at all so I was making some guesses in my changes